vendor/github.com/tonistiigi/fsutil/send_windows.go

```diff
@@ -0,0 +1,19 @@
+package fsutil
+
+import (
+	"context"
+
+	"github.com/Microsoft/go-winio"
+)
+
+func send(ctx context.Context, s *sender) error {
+	// Windows rootfs has a few special metadata files that
+	// require extra privileges to be accessed.
+
+	// TODO: [discussion] wrapping this from the caller's side in buildkit
+	// at the last point at session/filesync/diffcopy.go:25
+	// had no effect. Suspect is to do with the goroutines.
+	return winio.RunWithPrivileges([]string{winio.SeBackupPrivilege}, func() error {
+		return s.walk(ctx)

```

Comment:
Then as for the file skipping, also this needed to be at the place _walking_ is happening, which is in `fsutil`; I though of leaving that out so not to pollute the `fsutil` too much?

Right now those metadata files are being exported which I think it's okay, to have the end user deal with the files as they so wish?

```
    Directory: C:\dev\tmp\sample-1

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----            5/8/2021  9:43 AM                ProgramData
d----           5/31/2024  4:36 PM                System Volume Information <----
d----           5/10/2024 11:17 PM                Users
d----            6/7/2024 10:44 AM                WcSandboxState <----
d----           5/10/2024 11:17 PM                Windows
-a---            6/6/2024 12:44 PM             97 hello.txt
-a---           5/10/2024 11:14 PM           5647 License.txt
```

Reply:
The `WcSandboxState` should be skipped. It gets created by the host OS when mounting layers. It is of no use to users.

Reply:
Sure, the team was okay with skipping that, was listed among what's skipped in their layer exports:

```
"\\System Volume Information",
"\\ProgramData\\Microsoft\\Diagnosis",
"\\WcSandboxState",
```

Do you suggest adding that in the `fsutil` side or sending it from the caller?

Reply:
sending it from the caller makes more sense. If fsutil allows setting an exclusion list, that would be great.

Reply:
opened a separate issue for that here #5011